private/bufpkg/bufimage/bufimage.go

```diff
@@ -273,6 +273,19 @@ func WithNoReparse() NewImageForProtoOption {
 	}
 }
 
+// WithUnusedImportsComputation instructs NewImageForProto to compute unused imports
+// for the files. These are usually computed by the compiler and stored in the image.
```

Comment:
Is there room for us to do this in exactly one place throughout buf, without sacrificing performance? Ie could we in theory only do this here, and not do it in the compiler? Likely a dumb question but just asking

Reply:
Not realistically. The compiler has to traverse all edges of the graph anyway when resolving elements during linking and interpretation of options. So reporting unused imports is effectively free.

But after compilation is done, there is no way to get at this info other than to re-examine every edge again in the descriptor.

So if we turned it off in the compiler, with the idea of this being the "only place we do it": (1) other users of `protocompile` would no longer get the benefit of this warning and (2) we'd just be adding runtime overhead to compilation (to visit everything in the graph again) and not actually reduce the surface area of code to maintain (since the compiler must still visit everything in the graph to link).